### DISCUSSION TITLE
process.env.NODE_ENV undefined in next.config.js with custom server

### DISCUSSION THREAD

**Discussion by laclance** [vercel/next.js#48562] [Help]
### Summary

As soon as I started using a custom server process.env.NODE_ENV became undefined in next.config.js.

### Additional information

```js
Operating System:
      Platform: linux
      Arch: x64
      Version: #39~22.04.1-Ubuntu SMP PREEMPT_DYNAMIC Fri Mar 17 21:16:15 UTC 2
    Binaries:
      Node: 18.14.1
      npm: 9.5.0
      Yarn: 1.22.19
      pnpm: N/A
    Relevant packages:
      next: 13.3.0
      eslint-config-next: 13.3.0
      react: 18.2.0
      react-dom: 18.2.0
```

**Comment by haydenbleasel** [comment]
Hi @laclance, I believe this was covered a while ago here: https://github.com/vercel/next.js/issues/12269#issuecomment-791001193. Essentially you can use `@next/env` to utilize Next.js' env processing in a custom server. Let us know if it doesn't work though!

**Comment by laclance** [comment]
so I added ```const { loadEnvConfig } = require('@next/env');
loadEnvConfig('./', process.env.NODE_ENV !== 'production');```
to the top of my server.js but process.env.NODE_ENV is still undefined inside next.config.js

**Reply by haydenbleasel** [reply]
Hmm, you may want to implement the same code in your `next.config.js` seeing as that's where you want the environment variables.

**Reply by laclance** [reply]
thank you

**Reply by laclance** [reply]
oh wait sorry it does not work when running node ./server.js, only works with next dev
